src/github.com/matrix-org/go-neb/services/circleci/util.go

```diff
@@ -0,0 +1,61 @@
+package circleci
+
+import (
+	"unicode"
+	"unicode/utf8"
+)
+
+type buffer struct {
+	r         []byte
+	runeBytes [utf8.UTFMax]byte
+}
+
+func (b *buffer) write(r rune) {
+	if r < utf8.RuneSelf {
+		b.r = append(b.r, byte(r))
+		return
+	}
+	n := utf8.EncodeRune(b.runeBytes[0:], r)
+	b.r = append(b.r, b.runeBytes[0:n]...)
+}
+
+func (b *buffer) indent() {
+	if len(b.r) > 0 {
+		b.r = append(b.r, '_')
+	}
+}
+
+func CamelCaseToUnderscore(s string) string {
```

Comment:
this is part of the magic I'm not a fan of (obviously this bit is less magical but still not a fan)

Reply:
actually this whole file is for that magic